src/PSRule/Common/JsonConverters.cs

```diff
@@ -400,70 +401,75 @@ private IResource MapResource(JsonReader reader, JsonSerializer serializer)
+                    }
 
-                        // Read metadata
-                        else if (TryMetadata(
-                            reader: reader,
-                            serializer: serializer,
-                            propertyName: propertyName,
-                            metadata: out ResourceMetadata metadataValue))
-                        {
-                            metadata = metadataValue;
-                        }
+                    // Read metadata
+                    else if (TryMetadata(
+                        reader: reader,
+                        serializer: serializer,
+                        propertyName: propertyName,
+                        metadata: out ResourceMetadata metadataValue))
+                    {
+                        metadata = metadataValue;
+                    }
 
-                        // Try Spec
-                        else if (kind != null && TrySpec(
-                            reader: reader,
-                            serializer: serializer,
-                            propertyName: propertyName,
-                            apiVersion: apiVersion,
-                            kind: kind,
-                            metadata: metadata,
-                            comment: comment,
-                            spec: out IResource specValue))
-                        {
-                            result = specValue;
-                        }
+                    // Try Spec
+                    else if (kind != null && TrySpec(
+                        reader: reader,
+                        serializer: serializer,
+                        propertyName: propertyName,
+                        apiVersion: apiVersion,
+                        kind: kind,
+                        metadata: metadata,
+                        comment: comment,
+                        spec: out IResource specValue))
+                    {
+                        result = specValue;
 
-                        else
+                        // Break out of loop if result is populated
+                        // Needed so we don't read more than we have to
+                        if (result != null && reader.TokenType == JsonToken.EndObject)
                         {
-                            reader.Skip();
+                            break;
                         }
-
                     }
-                    else if (reader.TokenType == JsonToken.Comment)
+
+                    else
                     {
-                        var commentLine = reader.Value.ToString().TrimStart();
+                        reader.Skip();
+                    }
+                }
+
+                else if (reader.TokenType == JsonToken.Comment)
+                {
+                    var commentLine = reader.Value.ToString().TrimStart();
 
-                        if (commentLine.StartsWith("Synopsis: "))
+                    if (commentLine.StartsWith("Synopsis: "))
```

Comment:
Maybe add a length check here as well.

```suggestion
                    if (commentLine.Length > 10 && commentLine.StartsWith("Synopsis: "))
```